File: frontend/src/app/general.classes/Quote.ts
import { AccountingForm } from './AccountingForm';
import { Invoice } from './Invoice';
import { Customer } from './Customer';
import { Item } from './Item';

export class Quote extends AccountingForm {
  constructor(public id: string,
              public date: Date,
              public customerId: string,
              public items: Item[],
              public validUntil: Date,
              public invoiceId: string) {
                super(id, date, customerId, items);
              }
}

File: PlowNoMoreAPI/PlowNoMoreAPI/Classes/Quote.cs
using System;
using System.Collections.Generic;
namespace PlowNoMoreAPI.Classes
{
    public class Quote : AccountingFormBase
    {
        public DateTime validUntil { get; set; }
        public string invoiceId { get; set; }
        public Quote(string id, DateTime date, string customerId, List<LineItem> items, DateTime validUntil, string invoiceId)
            : base(id, date, customerId, items)
        {
            this.validUntil = validUntil;
            this.invoiceId = invoiceId;
        }
    }
}
